src/home_robot/home_robot/utils/morphology.py

```diff
@@ -49,3 +50,103 @@ def binary_closing(binary_image, kernel):
 
 def binary_denoising(binary_image, kernel):
     return binary_opening(binary_closing(binary_image, kernel), kernel)
+
+
+def get_edges(mask: torch.Tensor, threshold: float = 0.5) -> torch.Tensor:
+    """Extract edges from a torch tensor."""
+
+    mask = mask.float()
+
+    # Define the Sobel filter kernels
+    sobel_x = torch.tensor([[-1, 0, 1], [-2, 0, 2], [-1, 0, 1]], dtype=torch.float32)
```

Comment:
If `mask` is on gpu then this will cause an error unless the kernels are on the same device: i.e.
`sobel_x = torch.tensor([[-1, 0, 1], [-2, 0, 2], [-1, 0, 1]], dtype=torch.float32, device=mask.device)`

Reply:
ty